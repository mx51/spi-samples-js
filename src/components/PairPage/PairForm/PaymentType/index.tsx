import React from 'react';
import Box from '@material-ui/core/Box';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Grid from '@material-ui/core/Grid';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import Typography from '@material-ui/core/Typography';
import { SPI_PAIR_STATUS } from '../../../../definitions/constants/commonConfigs';
import { ReactComponent as FYROIcon } from '../../../../images/SP_logo_fyro.svg';
import { ReactComponent as LinkLinkIcon } from '../../../../images/SP_logo_linklink.svg';
import { ReactComponent as SPIIcon } from '../../../../images/SP_logo_SPI.svg';
import { ReactComponent as ZNVIcon } from '../../../../images/SP_logo_ZNV.svg';
import { useAppSelector } from '../../../../redux/hooks';
import { selectPairFormSerialNumber } from '../../../../redux/reducers/PairFormSlice/PairFormSelectors';
import { terminalInstance } from '../../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import useStyles from '../index.styles';

export default function PaymentType(): React.ReactElement {
  const classes = useStyles();
  // read redux store states
  const pairFormSerialNumber = useAppSelector(selectPairFormSerialNumber);
  const terminal = useAppSelector(terminalInstance(pairFormSerialNumber));

  return (
    <>
      <Grid item className={classes.title}>
        <Typography variant="h6" component="h1">
          Payment type
        </Typography>
      </Grid>
      <Grid container direction="column">
        <Grid container direction="row" className={classes.fieldSpace}>
          <FormControl className={classes.paymentTypeWrapper} component="fieldset" fullWidth>
            <RadioGroup aria-label="paymentType" name="paymentType" value="spi">
              <FormControlLabel
                className={classes.paymentTypeRadioButton}
                control={<Radio color="primary" />}
                disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
                label={
                  <Box display="flex" alignItems="center">
                    <SPIIcon className={classes.paymentTypeRadioButtonIcon} />
                    <span>Simple Payments Integration</span>
                  </Box>
                }
                value="spi"
              />
              <FormControlLabel
                className={classes.paymentTypeRadioButton}
                control={<Radio color="primary" />}
                disabled
                label={
                  <Box display="flex" alignItems="center">
                    <LinkLinkIcon className={classes.paymentTypeRadioButtonIcon} />
                    <span>Cloud Connect</span>
                  </Box>
                }
                value="link"
              />
              <FormControlLabel
                className={classes.paymentTypeRadioButton}
                control={<Radio color="primary" />}
                disabled
                label={
                  <Box display="flex" alignItems="center">
                    <ZNVIcon className={classes.paymentTypeRadioButtonIcon} />
                    <span>ZNV PayFast</span>
                  </Box>
                }
                value="znv"
              />
              <FormControlLabel
                className={classes.paymentTypeRadioButton}
                control={<Radio color="primary" />}
                disabled
                label={
                  <Box display="flex" alignItems="center">
                    <FYROIcon className={classes.paymentTypeRadioButtonIcon} />
                    <span>Fyro Payments</span>
                  </Box>
                }
                value="fyro"
              />
            </RadioGroup>
          </FormControl>
        </Grid>
      </Grid>
    </>
  );
}
