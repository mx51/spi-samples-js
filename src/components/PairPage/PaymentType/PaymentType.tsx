import React, { ChangeEvent } from 'react';
import Box from '@material-ui/core/Box';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Grid from '@material-ui/core/Grid';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import Typography from '@material-ui/core/Typography';

import { ReactComponent as CloudIcon } from '../../../images/Cloud.svg';
import { ReactComponent as SPIIcon } from '../../../images/SP_logo_SPI.svg';

import useStyles from './index.styles';
import { useAppSelector } from '../../../redux/hooks';
import { selectPairFormSerialNumber } from '../../../redux/reducers/PairFormSlice/PairFormSelectors';
import { terminalInstance } from '../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import { SPI_PAIR_STATUS } from '../../../definitions/constants/commonConfigs';

export enum PAYMENT_TYPE {
  SPI = 'spi',
  SPI_CLOUD = 'spiCloud',
}

type paymentTypeRadioProps = {
  handleChange: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
  value: PAYMENT_TYPE;
};

const PaymentTypeComponent: React.FC<paymentTypeRadioProps> = ({ handleChange, value }) => {
  const classes = useStyles();
  // read redux store states
  const pairFormSerialNumber = useAppSelector(selectPairFormSerialNumber);
  const terminal = useAppSelector(terminalInstance(pairFormSerialNumber));

  return (
    <>
      <Grid item className={classes.title}>
        <Typography variant="h6" component="h1">
          Integration type
        </Typography>
      </Grid>
      <Grid container direction="column">
        <Grid container direction="row" className={classes.fieldSpace}>
          <FormControl className={classes.paymentTypeWrapper} component="fieldset" fullWidth>
            <RadioGroup aria-label="paymentType" name="paymentType" value={value} onChange={handleChange}>
              <FormControlLabel
                className={classes.paymentTypeRadioButton}
                control={<Radio color="primary" />}
                disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
                label={
                  <Box display="flex" alignItems="center">
                    <SPIIcon className={classes.paymentTypeRadioButtonIcon} />
                    <span>Simple Payments Integration</span>
                  </Box>
                }
                value={PAYMENT_TYPE.SPI}
              />
              <FormControlLabel
                className={classes.paymentTypeRadioButton}
                control={<Radio color="primary" />}
                disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
                label={
                  <Box display="flex" alignItems="center">
                    <CloudIcon className={classes.paymentTypeRadioButtonIcon} />
                    <span>SPI Cloud</span>
                  </Box>
                }
                value={PAYMENT_TYPE.SPI_CLOUD}
              />
            </RadioGroup>
          </FormControl>
        </Grid>
      </Grid>
    </>
  );
};

export default PaymentTypeComponent;
