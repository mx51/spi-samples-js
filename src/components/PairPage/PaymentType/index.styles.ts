import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    title: {
      marginBottom: theme.spacing(1.5),
    },
    fieldSpace: {
      marginBottom: theme.spacing(1.5),
    },
    paymentTypeWrapper: {
      backgroundColor: theme.palette.common.white,
      borderRadius: theme.spacing(0.5),
      '&.MuiFormControl-root': {
        border: `1px solid ${theme.palette.secondary.main}`,
        marginTop: theme.spacing(1),
        padding: theme.spacing(0, 1),
      },
      '& .MuiFormControlLabel-root': {
        margin: theme.spacing(0, 2),
      },
    },
    paymentTypeRadioButton: {
      borderBottom: `1px solid ${theme.palette.secondary.main}`,
      padding: theme.spacing(1, 0),
      '&:last-child': {
        borderBottom: 'none',
      },
      '& .MuiRadio-root': {
        marginLeft: theme.spacing(-1.5),
      },
    },
    paymentTypeRadioButtonIcon: {
      maxHeight: '1.875rem',
    },
  })
);

export default useStyles;
