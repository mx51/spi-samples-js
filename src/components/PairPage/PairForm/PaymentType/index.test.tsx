import React from 'react';
import { cleanup } from '@testing-library/react';
import PaymentType from '.';
import mockWithRedux, { defaultMockTerminals, mockTerminalInstanceId } from '../../../../utils/tests/common';

describe('Test <PaymentType />', () => {
  afterEach(cleanup);

  test('should match PaymentType snapshot test', () => {
    // Arrange
    const terminal = defaultMockTerminals[mockTerminalInstanceId];
    const container = mockWithRedux(<PaymentType terminal={terminal} />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
