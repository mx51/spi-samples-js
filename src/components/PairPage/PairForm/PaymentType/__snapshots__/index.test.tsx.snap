// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PaymentType /> should match PaymentType snapshot test 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-title-3 MuiGrid-item"
  >
    <h1
      class="MuiTypography-root MuiTypography-h6"
    >
      Payment type
    </h1>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root makeStyles-fieldSpace-5 MuiGrid-container"
    >
      <fieldset
        class="MuiFormControl-root makeStyles-paymentTypeWrapper-12 MuiFormControl-fullWidth"
      >
        <div
          aria-label="paymentType"
          class="MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root makeStyles-paymentTypeRadioButton-13"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-17 Mui-checked MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-19"
                  name="paymentType"
                  type="radio"
                  value="spi"
                />
                <div
                  class="PrivateRadioButtonIcon-root-20 PrivateRadioButtonIcon-checked-22"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-21"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <div
                class="MuiBox-root MuiBox-root-23"
              >
                <svg
                  class="makeStyles-paymentTypeRadioButtonIcon-14"
                >
                  SP_logo_SPI.svg
                </svg>
                <span>
                  Simple Payments Integration
                </span>
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root makeStyles-paymentTypeRadioButton-13 Mui-disabled"
          >
            <span
              aria-disabled="true"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-disabled-18 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
              tabindex="-1"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-19"
                  disabled=""
                  name="paymentType"
                  type="radio"
                  value="link"
                />
                <div
                  class="PrivateRadioButtonIcon-root-20"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-21"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
            >
              <div
                class="MuiBox-root MuiBox-root-24"
              >
                <svg
                  class="makeStyles-paymentTypeRadioButtonIcon-14"
                >
                  SP_logo_linklink.svg
                </svg>
                <span>
                  Cloud Connect
                </span>
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root makeStyles-paymentTypeRadioButton-13 Mui-disabled"
          >
            <span
              aria-disabled="true"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-disabled-18 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
              tabindex="-1"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-19"
                  disabled=""
                  name="paymentType"
                  type="radio"
                  value="znv"
                />
                <div
                  class="PrivateRadioButtonIcon-root-20"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-21"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
            >
              <div
                class="MuiBox-root MuiBox-root-25"
              >
                <svg
                  class="makeStyles-paymentTypeRadioButtonIcon-14"
                >
                  SP_logo_ZNV.svg
                </svg>
                <span>
                  ZNV PayFast
                </span>
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root makeStyles-paymentTypeRadioButton-13 Mui-disabled"
          >
            <span
              aria-disabled="true"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-disabled-18 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
              tabindex="-1"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-19"
                  disabled=""
                  name="paymentType"
                  type="radio"
                  value="fyro"
                />
                <div
                  class="PrivateRadioButtonIcon-root-20"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-21"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
            >
              <div
                class="MuiBox-root MuiBox-root-26"
              >
                <svg
                  class="makeStyles-paymentTypeRadioButtonIcon-14"
                >
                  SP_logo_fyro.svg
                </svg>
                <span>
                  Fyro Payments
                </span>
              </div>
            </span>
          </label>
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;
